pkg/oadp/dataprotectionapplication.go

```diff
@@ -0,0 +1,367 @@
+		}
+
+		unsObject, err := builder.apiClient.Resource(
+			GetDataProtectionApplicationGVR()).Namespace(builder.Definition.Namespace).Create(
+			context.TODO(), &unstructured.Unstructured{Object: unstructuredDPA}, metav1.CreateOptions{})
+
+		if err != nil {
+			glog.V(100).Infof("Failed to create dataprotectionapplication")
+
+			return nil, err
+		}
+
+		builder.Object, err = builder.convertToStructured(unsObject)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return builder, err
+}
+
+// Update renovates the existing dataprotectionapplication object with
+// the dataprotectionapplication definition in builder.
+func (builder *DPABuilder) Update(force bool) (*DPABuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("failed to update dataprotectionapplication, object does not exist on cluster")
+	}
+
+	glog.V(100).Infof("Updating the dataprotectionapplication object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	if builder.errorMsg != "" {
+		return nil, fmt.Errorf(builder.errorMsg)
+	}
+
+	builder.Definition.ResourceVersion = builder.Object.ResourceVersion
+	builder.Definition.ObjectMeta.ResourceVersion = builder.Object.ObjectMeta.ResourceVersion
+
+	unstructuredDPA, err := runtime.DefaultUnstructuredConverter.ToUnstructured(builder.Definition)
+	if err != nil {
+		glog.V(100).Infof("Failed to convert structured DataProtectionApplication to unstructured object")
+
+		return nil, err
+	}
+
+	unstructObj, err := builder.apiClient.Resource(
+		GetDataProtectionApplicationGVR()).Namespace(builder.Definition.Namespace).Update(
+		context.TODO(), &unstructured.Unstructured{Object: unstructuredDPA}, metav1.UpdateOptions{})
+
+	if err != nil {
+		if force {
+			glog.V(100).Infof(
+				msg.FailToUpdateNotification("dataprotectionapplication", builder.Definition.Name, builder.Definition.Namespace))
+
+			err := builder.Delete()
+
+			if err != nil {
+				glog.V(100).Infof(
+					msg.FailToUpdateError("dataprotectionapplication", builder.Definition.Name, builder.Definition.Namespace))
+
+				return nil, err
+			}
+
+			return builder.Create()
+		}
+	}
+
+	if err == nil {
+		structuredDPA, err := builder.convertToStructured(unstructObj)
+		if err != nil {
+			glog.V(100).Infof("Failed to convert unstructured dataprotectionapplication into structured object")
+
+			return nil, err
+		}
+
+		builder.Object = structuredDPA
+	}
+
+	return builder, err
+}
+
+// Delete removes the dataprotectionapplication object and resets the builder object.
+func (builder *DPABuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the dataprotectionapplication object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	if !builder.Exists() {
+		glog.V(100).Infof("Dataprotection application %s does not exist in namespace %s",
```

Comment:
Can you please change log to:
dataprotectionapplication %s in namespace %s cannot be deleted because it does not exist